homeassistant/components/light/mqtt.py

```diff
@@ -234,12 +234,13 @@ def __init__(self, name, unique_id, effect_list, topic, templates,
         @callback
         def state_received(topic, payload, qos):
             """Handle new MQTT messages."""
-            payload = templates[CONF_STATE](payload)
-            if payload == self._payload['on']:
-                self._state = True
-            elif payload == self._payload['off']:
-                self._state = False
-            self.async_schedule_update_ha_state()
+            if templates[CONF_STATE](payload):
```

Comment:
Please only render each template once and use a guard clause:

```python
rendered = templates[CONF_STATE](payload)

if not rendered:
    return

…
```

Reply:
Also let's include a debug message if we ignore a message.

Reply:
OK, should be fixed now!